core/src/main/scala/org/apache/spark/h2o/H2OContext.scala

```diff
@@ -575,8 +575,10 @@ object H2OContext extends Logging {
       SharedBackendConf.PROP_REST_API_BASED_CLIENT._2)
     val isExternalBackend = conf.runsInExternalClusterMode
     if (isExternalBackend && isRestApiBasedClient) {
-      if (instantiatedContext.get() != null) {
-        if (connectingToNewCluster(instantiatedContext.get(), checkedConf)) {
+      val existingContext = instantiatedContext.get()
```

Comment:
This would definitely work, but I'm curios why the condition in the `connectingToNewCluster(existingContext, checkedConf)` failed.

The nodes should be the same when connecting to cluster created in automatic mode shouldn't it?

Reply:
If the `getOrCreate` method is called for the second time with a different instance of H2OConf, information about h2o cluster might be missing, since it's set once [here](https://github.com/h2oai/sparkling-water/blob/2f916d1e80e598a93f90c78d7d46715aca309605/core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalH2OBackend.scala#L211). The second call than fails on [this line](https://github.com/h2oai/sparkling-water/blob/282524b96d7972471a8fb79a54634f609d77c222/core/src/main/scala/org/apache/spark/h2o/utils/H2OContextRestAPIUtils.scala#L156).

Reply:
I see, very good catch!